### PR TITLE
fix: store message queue

### PR DESCRIPTION
**Pull request by gfanton**
In the event handler for an incoming message, simply put it in a message queue and return quickly. In a separate gorouting, process each message in the queue.

### CODE REVIEW

File: store_message.go [hunk 1]
```diff
@@ -136,7 +138,7 @@ func (m *MessageStore) processMessage(ctx context.Context, message *messageItem)
 	// process message
 	msg, err := m.secretStore.OpenEnvelopePayload(ctx, message.env, message.headers, m.groupPublicKey, m.currentDevicePublicKey, message.hash)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("unable to open the envelop: %w", err)
```
Comment (jefft0) [inline store_message.go:141]: "envelop" should be "envelope"

File: store_message.go [hunk 2]
```diff
@@ -153,78 +155,85 @@ func (m *MessageStore) processMessage(ctx context.Context, message *messageItem)
...
-
-			m.muDeviceCaches.Unlock()
+			// if we got any error here, put (back) the message into the device queue
+			// for ex: `too many open files` error
+			device.queue.Add(message)
+			_ = m.emitters.groupCacheMessage.Emit(*message)
 			continue
 		}
 
+		// if we get there we probably can process other messages (if any) in the device queue
```
Comment (jefft0) [inline store_message.go:191]: "get there" should be "get here"

File: store_message.go [hunk 3]
```diff
@@ -370,9 +378,10 @@ func constructorFactoryGroupMessage(s *WeshOrbitDB, logger *zap.Logger) iface.St
 		replication := false
 
 		store := &MessageStore{
+			// cmessage:       make(chan *messageItem),
```
Comment (jefft0) [inline store_message.go:381]: Remove this commented-out variable?

File: internal/queue/priority.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import (
+	"container/heap"
+	"sync"
+)
+
+type ICounter interface {
+	Counter() uint64
+}
+
+// A priorityMessageQueue implements heap.Interface and holds Items.
```
[the remark this answers is not in the document]
Comment (gfanton) [reply internal/queue/priority.go:12]: i renamed `messages` to `items`

**Pull request merged**